glass/points.py

```diff
@@ -279,7 +279,9 @@ def positions_from_delta(  # noqa: PLR0912, PLR0913, PLR0915
                 yield lon, lat, ipix.size * cmask
 
         # make sure that the correct number of pixels was sampled
-        assert np.sum(n[stop:]) == 0  # noqa: S101
+        if np.sum(n[stop:]) != 0:
+            msg = "The number of pixels sampled does not match the expected count."
+            raise ValueError(msg)
```

Comment:
This is also something that can only happen when something is intrinsically broken.

Is `S101` really a helpful rule if it flags instances where an assert is definitely the right tool for the job?

Reply:
I can revert though